js/controllers.js

```diff
@@ -1,12 +1,192 @@
+        var myChart = {};
+       
+        $scope.loadChartLibrary = function(libraryIndex) {   
+            myChart = libraryName[libraryIndex];            
+            switchChartLib(myChart.library);
+            $scope.setActive('b2',myChart.library);
+            $scope.b = true;
+            drawChart();
+            
+        };
+ 
+
+        //getDataing highchart sample 
+        $scope.chartConfig = highchart.chartConfig;
+
+        //getting nvd3 chart sample
+        $scope.options = ncd3Chart1.options;
+        $scope.data = ncd3Chart1.data;
+
+        //getting google chart sample
+        $scope.chart = googleChart;
+
+        //Switch all chart libaraies chart types at one
+        $scope.switchChartType = function(gchartType, hchartType, dchartType) {
+            $scope.chart.type = gchartType;
+            $scope.chartConfig.options.chart.type = hchartType;
+            $scope.options.chart.type = dchartType;
+            $scope.setActive('b3',hchartType);
+            //nvd3             
+            loadYAxisLabel($scope.fileName);
+            var nvdLabels = nvd3AxisLabels;
+            $scope.options.chart.xAxis = {
+                'axisLabel': 'Make'
+            };
+            $scope.options.chart.xAxis = {
+                'tickFormat': function(d) {
+                    return nvdLabels[d]
+                }
+            };
+
+        };
+
+        //deafult button picker
+        $scope.active={'b1':false,'b2':false,'b3':'bar'};
+
+        $scope.setActive = function(set,type) {
+            if(set=='b1')
+                $scope.active.b1 = type;
+            if(set=='b2')
+                $scope.active.b2 = type;
+            if(set=='b3')
+                $scope.active.b3 = type;
+        };
+
+        $scope.isBActive = function(set,type) {
+            if(set=='b1')
+              return type === $scope.active.b1;
+            if(set=='b2')
+                return type === $scope.active.b2;
+            if(set=='b3')
+                return type === $scope.active.b3;
+        };
+        
+        switchChartLib = function(chartLib) {
+            console.log('switchChartLib to '+chartLib);
+            $scope.highxChart = false;
+            $scope.googlexChart = false;
+            $scope.NVD3Chart = false;
+            $scope[chartLib] = true;
+        };
+
+        drawChart = function() {   
+            console.log('draw the chart '+myChart.library);
+            if(myChart.model != null) {           
+                renderChart(myChart.model, data);
+                            //nvd3 axis update
+            $scope.options.chart.xAxis = {
+                'tickFormat': function(d) {
+                    return nvd3AxisLabels[d]
+                }
+            };
+        }
+
+        };
+        var nvd3AxisLabels = {};
+
+        
+        //nvd3 chartting 
+        loadYAxisLabel = function(fileName) {
+            nvd3AxisLabels = getData(fileName).row(nvd3YaxisModel).get(getNVD3Yaxis);

```

Comment:
I don't get it: 
`nvd3AxisLabels = getData(fileName).row(nvd3YaxisModel).get(getNVD3Yaxis)`
but
`getNVD3Yaxis = function(error, rows) {nvd3AxisLabels = rows;}`
something is wrong here
also, function that has `get` in the name should return something by convention